src/screens/error/ErrorScreen.tsx

```diff
@@ -1,30 +1,63 @@
 import React from 'react'
 import { Linking } from 'react-native'
+import { WebViewMessageEvent } from 'react-native-webview/lib/WebViewTypes'
+
+import Minilog from '@cozy/minilog'
 
 import { CozyNotFoundPage } from '/components/webviews/CozyNotFoundPage'
 import { OfflinePage } from '/components/webviews/OfflinePage'
 import { SupervisedWebView } from '/components/webviews/SupervisedWebView'
 import { goBack } from '/libs/RootNavigation'
 
-const HTML = {
+const log = Minilog('ErrorScreen')
+
+/**
+ * Typings
+ */
+interface ErrorScreenProps {
+  route: { params: { type: string } }
+}
+
+interface Source {
+  html: ReturnType<typeof HTML[keyof typeof HTML]>
+}
+
+/**
+ * Implementations
+ */
+const HTML: Record<string, () => string> = {
   cozyNotFound: CozyNotFoundPage,
   offline: OfflinePage
 }
 
-const handlers = {
-  backButton: () => goBack(),
-  mailto: () => Linking.openURL('mailto:contact@cozycloud.cc')
+const handlers: Record<string, () => undefined | Promise<undefined>> = {
+  backButton: () => goBack() as undefined,
+  mailto: (): Promise<undefined> =>
+    Linking.openURL('mailto:contact@cozycloud.cc') as Promise<undefined>
 }
 
-const handleMessage = ({ nativeEvent: { data } }) =>
-  Object.keys(handlers).forEach(
-    eventId => data?.includes(eventId) && handlers[eventId]()
+const handleMessage = async (
+  event: WebViewMessageEvent
+): Promise<undefined[]> =>
+  await Promise.all(
+    Object.keys(handlers).map(async eventId => {
+      if (!event.nativeEvent.data.includes(eventId)) return undefined
+
+      try {
+        await handlers[eventId]()
+      } catch (error) {
+        log.error('Error while handling event', eventId, error)
```

Comment:
Are we sure we want to silently catch ALL events? Here our need is to silently catch only the `mailto` one.

Reply:
Updated so now the event has to declare if it wants to be caught or not. I set `backButton` to `false` and `mailto` to `true`

Reply:
Why not just moving the try-catch in the handler that needs it?